[GEOT-5841] Fix "addFeatures(FeatureCollection)" to iterate over features

emerkle826 opened the pull request:
This should address https://osgeo-org.atlassian.net/browse/GEOT-5841.

aaime commented on modules/library/data/src/test/java/org/geotools/data/memory/MemoryDataStoreTest.java:
```diff
@@ -161,6 +161,10 @@ public void testMemoryDataStore() throws Exception {
     public void testMemoryDataStoreFeatureCollection() throws IOException {
         DataStore store = new MemoryDataStore(DataUtilities.collection(roadFeatures));
```
Looking at it, it did not seem a valid test as there is no call to addFeatures. I see the constructor calls in internally, but this could change in the future. Maybe add an explicit test too?

emerkle826 replied:
I can add an explicit one for addFeatures as well.

emerkle826 commented:
@aaime Thank you for the feedback. I've reverted the original test (in case the implicit functionality changes in the future) and added a separate explicit test for `addFeatures(FeatureCollection)`.

tbarsballe commented:
Does this need to be backported?

emerkle826 commented:
@tbarsballe @aaime This can be backported to every version of GeoTool from 16.0 and up. GeoTools 15.x does not have this issue. I'm sure some versions of GeoTools since 16 are no longer supported, so please backport to as few or as many as you wish.

tbarsballe commented:
Backported to 18.x and 17.x.
